tests/cases/fourslash/getJavaScriptSemanticDiagnostics23.ts

```diff
@@ -0,0 +1,14 @@
+/// <reference path="fourslash.ts" />
+
+// @allowJs: true
+// @Filename: a.js
+//// function Person(age) {
+////     if (age >= 18) {
+////         this.canVote = true;
+////     } else {
+////         this.canVote = false;

```

Comment:
What is the resulting type if the assignments are of different (incompatible) types (e.g. a boolean and an arrow function)? Do you get any errors or intellisense off the resulting type?

It would be good to add a test to cover that scenario for both documentation and regression purposes.